packages/sanity/src/core/form/studio/FormBuilder.tsx

```diff
@@ -306,10 +308,16 @@ function RootInput(props: RootInputProps) {
   const {rootInputProps, onPathOpen, openPath, renderInput} = props
   const treeEditing = useTreeEditingEnabled()
 
+  const open = useMemo(
+    () => shouldArrayDialogOpen(rootInputProps.schemaType, openPath),
+    [openPath, rootInputProps.schemaType],
+  )
+
```

Comment:
I think it makes more sense to conditionally render the dialog with this value here, rather than inside the dialog component itself.